subworkflows/local/alignment.nf

```diff
@@ -86,3 +90,32 @@ workflow ALIGNMENT {
         salmon_info     = SALMON_QUANT.out.json_info
         versions        = ch_versions
 }
+
+
+// Custom functions
+
+/**
+* Branch the read channel into differnt channels,
+* depending on whether the sample has multiple fastq files or not.
+* The resulting channels gets the original sample id in meta.
+*
+* @param ch_reads Channel containing meta and fastq reads
+* @return Channel containing meta with original id and branched on number of fastq files
+*/
+def branchFastqToSingleAndMulti(ch_reads) {
+
+    return ch_reads
+        .map {
+            meta, fastq ->
+                original_id = meta.id.split('_T')[0..-2].join('_')
+                [ meta + [id: original_id], fastq ]
+        }
+        .groupTuple()
+        .branch {
+            meta, fastq ->
+                single_fq: fastq.size() == 1
+                    return [ meta, fastq.flatten() ]
+                multiple_fq: fastq.size() > 1
+                    return [ meta, fastq.flatten() ]
```

Comment:
Is branching needed here? Asking because the returned output looks the same for both branches.

Reply:
https://github.com/nf-core/rnaseq/blob/3bec2331cac2b5ff88a1dc71a21fab6529b57a0f/workflows/rnaseq.nf#L211

Reply:
Alright cool! :D